src/m365/entra/commands/app/app-permission-add.ts

```diff
@@ -137,13 +143,45 @@ class EntraAppPermissionAddCommand extends GraphCommand {
     }
   }
 
+  private async getAppObjectId(appName: string): Promise<string> {
+    const requestOptions: CliRequestOptions = {
+      url: `${this.resource}/v1.0/myorganization/applications?$filter=displayName eq '${formatting.encodeQueryParameter(appName)}'&$select=id`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    const res = await request.get<{ value: { id: string }[] }>(requestOptions);
+
+    if (res.value.length === 1) {
+      return res.value[0].id;
+    }
+
+    if (res.value.length === 0) {
+      throw `No Entra application registration with name ${appName} found`;
+    }
+
+    const resultAsKeyValuePair = formatting.convertArrayToHashTable('id', res.value);
+    const result = await cli.handleMultipleResultsFound<{ id: string }>(`Multiple Entra application registration with name '${appName}' found.`, resultAsKeyValuePair);
+    return result.id;
+  }
+
   private async getAppObject(options: Options): Promise<Application> {
-    const apps = options.appObjectId
-      ? await odata.getAllItems<Application>(`${this.resource}/v1.0/applications?$filter=id eq '${options.appObjectId}'&$select=id,appId,requiredResourceAccess`)
-      : await odata.getAllItems<Application>(`${this.resource}/v1.0/applications?$filter=appId eq '${options.appId}'&$select=id,appId,requiredResourceAccess`);
+    let appObjectId = options.appObjectId ?? options.appId;
+    let appNotFoundMessage = `${options.appObjectId ? 'object id' : 'client id'} ${options.appObjectId ? options.appObjectId : options.appId}`;
+
+    if (options.appName) {
+      appObjectId = await this.getAppObjectId(options.appName);
```

Comment:
To fix the previous comment about the extra http call, you could just take care that everything is retrieved there using a `getAppObjectByName` function and return the value here.